### DISCUSSION TITLE
Augmentation for Object Detection pipeline

### DISCUSSION THREAD

**Discussion by pranavjadhav001** [keras-team/keras-cv#277] [General]
Going through the repo, I saw mostly image preprocessing and augmentation layers which allows only image transformations. Is there any scope for augmentation transformation of image and bbox for object detection like albumentations library. https://albumentations.ai/docs/getting_started/bounding_boxes_augmentation/ 
I also see image preprocessing operations have been created as layers to make it more as 1 single coherent model. Not sure, how the support must be added for object detection preprocessing layers where the input is image and labels.

**Comment by innat** [comment]
@pranavjadhav001 augmentation for obj detection will be added soon. Feel free to contribute.

**Reply by pranavjadhav001** [reply]
Thanks for the reply. I would love to contribute. Is it planned in the roadmap or will be added later?

**Reply by innat** [reply]
You can read the [roadmap](https://github.com/keras-team/keras-cv/blob/master/.github/ROADMAP.md#areas-of-focus).

**Reply by innat** [reply]
and design [proposal](https://github.com/keras-team/governance/blob/master/rfcs/20200827-keras-cv-scoping-design.md#design-proposal).

**Reply by LukeWood** [reply]
Thanks for linking this @innat !  I have just checked out this conversation.  Yes, we will be adding bbox support as one of our next big ongoing efforts!  @divyashreepathihalli is taking on a lot of this, but feel free to pick up layers and add an `augment_bounding_box` method similar to the `augment_image` and `augment_label` calls!

**Comment by jbischof** [comment]
This is done.
